src/overlays/actors/ovl_Dm_Char03/z_dm_char03.c

```diff
@@ -32,34 +32,155 @@ const ActorInit Dm_Char03_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char03/func_80AAB700.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char03/func_80AAB710.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char03/func_80AAB838.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char03/DmChar03_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char03/func_80AAB974.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char03/func_80AAB990.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char03/func_80AAB9A8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char03/DmChar03_Draw.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char03/func_80AABA84.s")
+AnimationInfo sAnimations[] = { { &object_osn_Anim_020530, 1.0f, 0.0f, -1.0f, ANIMMODE_ONCE, 0.0f } };
+
+void DmChar03_ChangeAnimation(SkelAnime* skelAnime, AnimationInfo* animInfo, u16 index) {
+    f32 frame;
+
+    animInfo += index;
+
+    if (animInfo->frameCount < 0.0f) {
+        frame = Animation_GetLastFrame(animInfo->animation);
+    } else {
+        frame = animInfo->frameCount;
+    }
+    Animation_Change(skelAnime, animInfo->animation, animInfo->playSpeed, animInfo->startFrame, frame, animInfo->mode,
+                     animInfo->morphFrames);
+}
+
+void DmChar03_Init(Actor* thisx, PlayState* play) {
+    DmChar03* this = THIS;
+
+    this->animIndex = 0;
+    this->actor.targetArrowOffset = 3000.0f;
+    this->unk_18E = false;
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 24.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gDekuMaskSkel, NULL, NULL, NULL, 0);
+    DmChar03_ChangeAnimation(&this->skelAnime, sAnimations, 0);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = DmChar03_DoNothing;
+}
+
+void DmChar03_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_80AAB5F8(DmChar03* this, PlayState* play) {
+    s32 index = Object_GetIndex(&play->objectCtx, OBJECT_GI_NUTSMASK);
+
+    if (index >= 0) {
+        this->objectIndex = index;
+        this->actionFunc = func_80AAB644;
+    }
+}
+
+void func_80AAB644(DmChar03* this, PlayState* play) {
+    if (Cutscene_CheckActorAction(play, 136)) {
+        s32 index = Cutscene_GetActorActionIndex(play, 136);
+        if (play->csCtx.actorActions[index]->action == 4) {
+            this->unk_18E = true;
+            this->offset.x = play->csCtx.actorActions[index]->startPos.x;
+            this->offset.y = play->csCtx.actorActions[index]->startPos.y;
+            this->offset.z = play->csCtx.actorActions[index]->startPos.z;
+        }
+    }
+}
+
+void DmChar03_DoNothing(DmChar03* this, PlayState* play) {
+}
+
+void func_80AAB710(DmChar03* this, PlayState* play) {
+    u8 var_a2 = true;
+
+    if (Cutscene_CheckActorAction(play, 0x88)) {
+        s32 index = Cutscene_GetActorActionIndex(play, 0x88);
+        if (play->csCtx.frames == play->csCtx.actorActions[index]->startFrame) {
```

Comment:
```suggestion
        s32 index = Cutscene_GetActorActionIndex(play, 0x88);

        if (play->csCtx.frames == play->csCtx.actorActions[index]->startFrame) {
```

Reply:
Ah my bad Chloe, slipped my mind to space after the index move.